crates/bevy_ecs/src/system/query.rs

```diff
@@ -1121,8 +1131,15 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> Query<'w, 's, Q, F> {
 
     /// Returns a mutable reference to the component `T` of the given entity.
     ///
+    /// Unlike [`Query::get_mut`], this method returns only a single component,
+    /// rather than the entire query item declared in the first type parameter.
+    ///
+    /// Note that the component must both be present on the entity and the access declared on the query,
+    /// via the first "fetch" type parameter.
     /// In case of a nonexisting entity or mismatched component, a [`QueryComponentError`] is returned instead.
     ///
+    /// If the declared access for the provided component is read-only, only an immutable reference will be returned.
```

Comment:
I don't think that's true for `get_component_mut`, it returns a `MissingWriteAccess` in that case, there is a doc test for that case: https://docs.rs/bevy/latest/bevy/ecs/system/enum.QueryComponentError.html#variant.MissingWriteAccess